agent/engine/dependencygraph/graph.go

```diff
@@ -367,6 +372,15 @@ func containerOrderingDependenciesIsResolved(target *apicontainer.Container,
 	}
 }
 
+func hasDependencyTimedOut(dependOnContainer *apicontainer.Container, dependencyCondition string) bool {
+	switch dependencyCondition {
+	case successCondition, completeCondition, healthyCondition:
+		return time.Now().After(dependOnContainer.GetStartedAt().Add(dependOnContainer.GetStartTimeout()))
```

Comment:
what happens when `GetStartedAt` is not set?

Reply:
If we call this after "resolves" then this shouldn't happen. But more defensive programming is a good idea too.

Reply:
Thanks Sharanya and Derek for the useful inputs. I have added both the checks.